File: BackendAhorroUy/WebApiModels/DataTypes/MarketDTs/BestOptionResponse.cs
using System;
namespace WebApiModels.DataTypes.MarketDTs
{
    public class BestOptionResponse
    {
        public string MarketName { get; set; }

        public string MarketAddress { get; set; }

        public double PriceForProducts { get; set; }

        public float MarketLongitude { get; set; }

        public float MarketLatitude { get; set; }

        public DateTime OpenTimeToday { get; set; }

        public DateTime CloseTimeToday { get; set; }

        public double Difference { get; set; }

        public string MarketLogo { get; set; }

        public void CalculateDifferenceToCloseInHours()
        {
            Difference = (CloseTimeToday - DateTime.Now).TotalHours;
        }

        public override bool Equals(object obj)
        {
            return obj is BestOptionResponse response &&
                   MarketName == response.MarketName &&
                   MarketAddress == response.MarketAddress &&
                   PriceForProducts == response.PriceForProducts &&
                   MarketLongitude == response.MarketLongitude &&
                   MarketLatitude == response.MarketLatitude;
        }

        public BestOptionResponse()
        {
        }
    }
}

File: AhorroUY/app/src/main/java/com/ahorrouy/api/model/marketMap/BestOptionResponse.kt
package com.ahorrouy.api.model.marketMap

import com.google.gson.annotations.SerializedName

data class BestOptionResponse(
    @SerializedName("marketName")
    val marketName: String,
    @SerializedName("marketAddress")
    val marketAddress: String,
    @SerializedName("marketLogo")
    val marketLogo: String,
    @SerializedName("priceForProducts")
    val priceForProducts: Double,
    @SerializedName("marketLongitude")
    val marketLongitude: Float,
    @SerializedName("marketLatitude")
    val marketLatitude: Float,
    @SerializedName("openTimeToday")
    val openTimeToday: String,
    @SerializedName("closeTimeToday")
    val closeTimeToday: String,
    @SerializedName("difference")
    val difference: Double
)
